File: strategy_kotlin/src/main/kotlin/strategy_training/JumpTraining.kt
package com.kmmania.strategy_training

class JumpTraining(private val trainingSession: TrainingSession) : AthleticTraining {
    override fun train() {
        println("*** Jump training ***")
        println("Date: ${trainingSession.date}")
        println("Time slot: ${trainingSession.timeSlot}")
        println("Place: ${trainingSession.place}")
    }
}

File: strategy_python/src/strategy_training/JumpTraining.py
from src.strategy_training.AthleticTraining import AthleticTraining


class JumpTraining(AthleticTraining):
    def __init__(self, training_session):
        self.training_session = training_session

    def train(self):
        print("*** Jump training ***")
        print(f"Date: {self.training_session.date}")
        print(f"Time slot: {self.training_session.time_slot}")
        print(f"Place: {self.training_session.place}")
